### PR TITLE
Support berlin fork

### PR DESCRIPTION
**Pull request by norhh**

### CODE REVIEW

File: mythril/laser/ethereum/state/machine_state.py [hunk 1]
```diff
@@ -110,6 +111,7 @@ def __init__(
         """
         self._pc = pc
         self.stack = MachineStack(stack)
+        self.rstack = MachineStack(rstack)
```
Comment (JoranHonig) [inline mythril/laser/ethereum/state/machine_state.py:114]: What do you  think about using a more explicit name? `rstack` might not be super clear for new users/ us in a few months
```suggestion
        self.subroutine_stack = MachineStack(subroutine_stack)
```

File: mythril/laser/ethereum/svm.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ def execute_state(
             hook(global_state)
 
         instructions = global_state.environment.code.instruction_list
-
+        print(instructions)
```
Comment (JoranHonig) [inline mythril/laser/ethereum/svm.py:316]: remove debug statement
```suggestion
```

File: mythril/laser/ethereum/instructions.py [hunk 3]
```diff
@@ -1629,7 +1666,7 @@ def pc_(self, global_state: GlobalState) -> List[GlobalState]:
         program_counter = global_state.environment.code.instruction_list[index][
             "address"
         ]
-        global_state.mstate.stack.append(program_counter)
+        global_state.mstate.stack.append(symbol_factory.BitVecVal(program_counter, 256))
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py:1669]: 🔥 good catch!

File: mythril/laser/ethereum/instructions.py [hunk 4]
```diff
@@ -1618,6 +1617,44 @@ def jumpi_(self, global_state: GlobalState) -> List[GlobalState]:
...
+            raise VmException("Encountered symbolic JUMPSUB location")
+        index = util.get_instruction_index(disassembly.instruction_list, location)
+        instr = disassembly.instruction_list[index]
+
+        if instr["opcode"] != "BEGINSUB":
+            raise VmException(
+                "Encountered invalid JUMPSUB location :{}".format(instr["address"])
+            )
+        global_state.mstate.rstack.append(
+            symbol_factory.BitVecVal(global_state.mstate.pc + 1, 256)
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py:1645]: Does this need to be a BitVecVal? The rstack is a stack of always concrete instruction offsets right?

**Review by JoranHonig — CHANGES_REQUESTED** [review]
Looking good 🎉 

Some minor comments

File: mythril/laser/ethereum/instructions.py [hunk 5]
```diff
@@ -1618,6 +1617,42 @@ def jumpi_(self, global_state: GlobalState) -> List[GlobalState]:
...
+        global_state.mstate.subroutine_stack.append(global_state.mstate.pc + 1)
+        global_state.mstate.pc = location
+        return [global_state]
+
+    @StateTransition(increment_pc=False)
+    def returnsub_(self, global_state: GlobalState) -> List[GlobalState]:
+        """
+        Returns control to the caller of the subroutine
+        """
+        global_state.mstate.pc = global_state.mstate.subroutine_stack.pop()
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py:1653]: What should happen if there is no element on the subroutine stack?

Comment (norhh) [reply mythril/laser/ethereum/instructions.py:1653]: The subroutine stack is a MachineStack(as it's properties are similar to EVM stack) which raises StackUnderflow Exception

**Review by JoranHonig — APPROVED** [review]

**Pull request merged**